src/server/raft_client/fresh.rs

```diff
@@ -682,10 +705,14 @@ async fn start<S, R>(
         }
         match res {
             Ok(()) => {
-                error!("connection fail"; "store_id" => back_end.store_id, "addr" => addr, "error" => "require fallback even with legacy API");
+                error!("connection fail"; "store_id" => back_end.store_id, "addr" => addr, "err" => "require fallback even with legacy API");
             }
             Err(_) => {
                 error!("connection abort"; "store_id" => back_end.store_id, "addr" => addr);
+                if retry_times > 1 {
+                    // Clears pending messages to avoid consuming high memory when one node is shutdown.
+                    back_end.clear_pending_message();
```

Comment:
Is the under `broadcast_unreachable` necessary? For all dropped messages `report_unreachable` is called.

Reply:
Yes. For example, it can make hibernated peers to wake up and campaign even though there are no messages sent.